src/services/rfqm_service.ts

```diff
@@ -699,6 +703,32 @@ export class RfqmService {
             ]);
         }
 
+        // verify that there is not a pending transaction for this taker and taker token
+        const pendingJobs = await this._dbUtils.findJobsWithStatusesAsync([
```

Comment:
Just curious, is it safe to do `NOT [PendingConfirmed, Confirmed]` instead? feels like easier to maintain?

Reply:
I had a thought about this, what if we created a constant called `PENDING_STATUSES`:

```
const PENDING_STATUSES = [
            RfqmJobStatus.PendingEnqueued,
            RfqmJobStatus.PendingProcessing,
            RfqmJobStatus.PendingLastLookAccepted,
            RfqmJobStatus.PendingSubmitted,
];
```

And then we could re-use this in multiple places. There are of course, other ways of declaring PENDING_STATUSES that might make it easier to maintain as well (filter for all statuses that have the word "pending")

Reply:
Honestly I hate this. The solution to handle this complexity is not add another layer of abstraction.

I actually have started to see this pattern a lot across the codebase and I'd like to call attention to it -- instead of duplicating a small amount of code an engineer chooses to create a new abstraction. The abstraction is then abstracted again, and maybe again; the original meaning getting buried deeper and deeper over the ages. All decisions feel appropriate at the time, and maybe are. But when one reads the resulting code it feels like you're reading a book and you've stuck a finger in three or four different parts of the book and you keep needing to flip back and forth between them all to get the full understanding of whatever you're looking at. You've made an unconscious decision to trade "easy to maintain" for "hard to understand".

We had a rule of thumb in our heads at my previous team working for a social media company for this and I think it served us well. Copy once, ok. Copy twice, think about it. Copy three times, time to refactor.

Reply:
> Just curious, is it safe to do `NOT [PendingConfirmed, Confirmed]` instead? feels like easier to maintain?

Those are not the only remaining statuses lol